gen/bind_generator.jl

```diff
@@ -6,12 +6,14 @@ const bind_exceptions = Dict{Symbol,Symbol}()
 
 # Version-numbered identifiers
 push!(bind_exceptions, :h5a_create => :H5Acreate2)
+push!(bind_exceptions, :h5a_iterate => :H5Aiterate2)
 push!(bind_exceptions, :h5d_create => :H5Dcreate2)
 push!(bind_exceptions, :h5d_open => :H5Dopen2)
 push!(bind_exceptions, :h5e_get_auto => :H5Eget_auto2)
 push!(bind_exceptions, :h5e_set_auto => :H5Eset_auto2)
 push!(bind_exceptions, :h5g_create => :H5Gcreate2)
 push!(bind_exceptions, :h5g_open => :H5Gopen2)
+push!(bind_exceptions, :h5l_iterate => :H5Literate1)
```

Comment:
If we get rid of this, will this work on v1.12 ? The documentation make it sound like it should still work, but it might depend on how the underlying libraries were built

https://confluence.hdfgroup.org/display/HDF5/H5L_ITERATE
> The function H5L_ITERATE was renamed to H5L_ITERATE1 and deprecated in this release.The macro H5L_ITERATE and the function H5L_ITERATE2 were introduced in this release.
--

Reply:
The problem is they implement the rename as C header `#define`s, so the symbol doesn't really exist in the library.
```julia
julia> dlsym(HDF5_jll.libhdf5_handle, :H5Literate)
ERROR: could not load symbol "H5Literate":
/home/justin/.julia/artifacts/997813d46a8a06e6e9871a2a01483f91ce954eca/lib/libhdf5.so: undefined symbol: H5Literate
Stacktrace:
 [1] #dlsym#1
   @ ./libdl.jl:56 [inlined]
 [2] dlsym(hnd::Ptr{Nothing}, s::Symbol)
   @ Base.Libc.Libdl ./libdl.jl:54
 [3] top-level scope
   @ REPL[27]:1

julia> dlsym(HDF5_jll.libhdf5_handle, :H5Literate1)
Ptr{Nothing} @0x00007ff367429980
```

Reply:
Darn it...

Reply:
So I've done some more generator macro hacking to try to work around this.

Second commit makes the generator a bit smarter in dealing with versioned symbols so that there are only a few exceptions in the `bind_exceptions` Dict.

The third commit is what actually does the version-specific hacking. I'm not convinced yet that there isn't a smarter way to go about this, but this is what came to mind and I could make work.